AdyenCard/Form/FormCardNumberItem.swift

```diff
@@ -49,21 +49,20 @@ internal final class FormCardNumberItem: FormTextItem {
     
     internal func valueDidChange() {
         binValue = String(value.prefix(FormCardNumberItem.binLength))
+        cardNumberFormatter.cardType = supportedCardTypes.adyen.type(forCardNumber: value)
         
         guard !value.isEmpty else {
-            cardNumberFormatter.cardType = nil
             previouslyDetectedCardTypes = nil
             setLogos(for: Set<CardType>(supportedCardTypes))
             return
         }
     }
     
-    internal func detectedCardsDidChange(detectedCards: [CardType]) {
+    internal func didChange(detectedCards: [CardType]) {
         let newDetectedCardTypes = Set<CardType>(detectedCards)
         guard self.previouslyDetectedCardTypes != newDetectedCardTypes else { return }
         
         previouslyDetectedCardTypes = newDetectedCardTypes
-        cardNumberFormatter.cardType = newDetectedCardTypes.first
```

Comment:
why remove this line? the formatter should adjust to the true card types detected by the backend, right?

Reply:
It moved to line 52. `cardNumberFormatter` is mostly detecting Amex / no-Amex, so local regex-based recognition is more appropriate.

Reply:
👍 ok